pkg/skaffold/deploy/kpt/kpt.go

```diff
@@ -116,15 +124,12 @@ func versionCheck(dir string) error {
 		re := regexp.MustCompile(kustomizeVersionRegexP)
 		match := re.FindStringSubmatch(versionInfo)
 		if len(match) != 2 {
-			return fmt.Errorf("unknown kustomize version %v\nPlease upgrade your "+
```

Comment:
WDYT about moving this kustomize version parsing logic into the `kustomize` package, similar to how we do `kubectl.CompareVersionTo()` checks?   And similarly extracting the `kpt` version checking too?  Then the version-checking can be tested in isolation and not involve string matching against the warning text.

Reply:
Very reasonable suggestion. We can add similar logic to kustomize package, but we may want to separate the version check from kpt. This is because we will replace the kustomize CLI to builtin kustomize package in kpt. This is pending until the unified kpt/kustomize design is done in the kpt team.